Documentation/gettingstarted/external-workloads.rst

```diff
@@ -0,0 +1,353 @@
+  each other.
+
+* The network between the external workloads and your cluster must
+  allow the node-cluster communication. The exact ports are documented
+  in the :ref:`firewall_requirements` section.
+
+
+Prepare your cluster
+####################
+
+Enable support for external workloads
+=====================================
+
+.. include:: k8s-install-download-release.rst
+
+Your cluster must be configured with support for external workloads
+enabled. This can also be done by passing ``--set
+externalWorkloads.enabled=true`` to ``helm install`` when installing
+or updating Cilium:
+
+.. parsed-literal ::
+
+    helm install cilium |CHART_RELEASE|         \\
+      --namespace kube-system                   \\
+      --set externalWorkloads.enabled=true
+
+This will add a deployment for ``clustermesh-apiserver`` into your
+cluster, as well as the related cluster resources, such as TLS
+secrets.
+
+.. _gs_external_workloads_tls_configuration:
+
+TLS configuration
+=================
+
+By default TLS secrets are created by helm. This has the downside that
+each time you run ``helm`` the TLS secrets will be re-created, forcing
+each external workload to be reconfigured to be able to connect to
+your cluster. There are two ways to get around this. Firstly, you can
+enable a cluster job to create TLS secrets instead. This way the CA
+cert secrets are reused as long as they are not removed from your
+cluster:
+
+.. parsed-literal ::
+
+    helm install cilium |CHART_RELEASE|                     \\
+      --namespace kube-system                               \\
+      --set externalWorkloads.enabled=true                  \\
+      --set clustermesh.apiserver.tls.auto.method=cronJob
+
+Secondly, you can use your own CA certs. You can either create the
+``clustermesh-apiserver-ca-cert`` secret in your Cilium install
+namespace (e.g., ``kube-system``) yourself, or pass the CA cert and
+key to helm:
+
+.. parsed-literal ::
+
+    helm install cilium |CHART_RELEASE|                     \\
+      --namespace kube-system                               \\
+      --set externalWorkloads.enabled=true                  \\
+      --set clustermesh.apiserver.tls.auto.method=cronJob   \\
+      --set clustermesh.apiserver.tls.ca.cert="<< base64 encoded CA certificate >>"   \\
+      --set clustermesh.apiserver.tls.ca.key="<< base64 encoded CA key >>"
+
+You can also allow the cronJob create the secrets and then save the
+generated CA certs to be used later:
+
+.. code:: bash
+
+    kubectl -n kube-system get secret clustermesh-apiserver-ca-cert -o yaml > clustermesh-apiserver-ca-cert.yaml
+   
+.. _gs_external_workloads_expose_clustermesh_apiserver:
+
+Expose cluster to external workloads
+====================================
+
+``clustermesh-apiserver`` must be exposed to the external
+workloads. By default this is done using a HostPort service on
+port 32379. It is also possible to use the ``LoadBalancer`` service
```

Comment:
Is it a HostPort or a NodePort? The first tab below says NodePort.

Reply:
NodePort, fixed.